src/librustc_trans/mir/block.rs

```diff
@@ -798,46 +853,46 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
         let slot = self.get_personality_slot(&bcx);
         bcx.store(llretval, slot);
         bcx.br(target.llbb());
-        block
+        bcx.llbb()
     }
 
-    pub fn init_cpad(&mut self, bb: mir::BasicBlock) {
-        let bcx = self.bcx(bb);
+    pub fn init_cpad(&mut self, bb: mir::BasicBlock,
+        funclets: &mut IndexVec<mir::BasicBlock, Option<Funclet>>) {
+        let bcx = self.build_block(bb);
```

Comment:
This function can probably just iterate over `cleanup_kinds` to produce `funclets`, without the RPO visit order (as `CleanupKind::Internal` doesn't need to reference a different `functlets` entry).